acceptance/detector_test.go

```diff
@@ -0,0 +1,197 @@
+			groupContents, err := ioutil.ReadFile(tempGroupToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(groupContents) > 0, true)
+			var buildpackGroup lifecycle.BuildpackGroup
+			_, err = toml.Decode(string(groupContents), &buildpackGroup)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildpackGroup.Group[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildpackGroup.Group[0].Version, "simple_buildpack_version")
+
+			// check plan.toml - should be empty
+			tempPlanToml := filepath.Join(copyDir, "layers", "plan.toml")
+			planContents, err := ioutil.ReadFile(tempPlanToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(planContents) > 0, true)
+			var buildPlan lifecycle.BuildPlan
+			_, err = toml.Decode(string(planContents), &buildPlan)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildPlan.Entries[0].Providers[0].ID, "simple_buildpack")
+			h.AssertEq(t, buildPlan.Entries[0].Providers[0].Version, "simple_buildpack_version")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Name, "some_requirement")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Metadata["some_metadata_key"], "some_metadata_val")
+			h.AssertEq(t, buildPlan.Entries[0].Requires[0].Metadata["version"], "0.1")
+		})
+	})
+
+	when("environment variables are provided for buildpack and app directories and for the output files", func() {
+		var copyDir, containerName string
+
+		it.Before(func() {
+			containerName = "test-container-" + h.RandString(10)
+			var err error
+			copyDir, err = ioutil.TempDir("", "test-docker-copy-")
+			h.AssertNil(t, err)
+		})
+
+		it.After(func() {
+			if h.DockerContainerExists(t, containerName) {
+				h.Run(t, exec.Command("docker", "rm", containerName))
+			}
+			os.RemoveAll(copyDir)
+		})
+
+		it("writes group.toml and plan.toml in the right location and with the right names", func() {
+			h.DockerRunAndCopy(t,
+				containerName,
+				copyDir,
+				detectImage,
+				"/layers",
+				h.WithFlags("--user", userID,
+					"--env", "CNB_ORDER_PATH=/cnb/orders/always_detect_order.toml",
+					"--env", "CNB_BUILDPACKS_DIR=/cnb/custom_buildpacks",
+					"--env", "CNB_APP_DIR=/custom_workspace",
+					"--env", "CNB_GROUP_PATH=./custom_group.toml",
+					"--env", "CNB_PLAN_PATH=./custom_plan.toml",
+					"--env", "CNB_PLATFORM_DIR=/custom_platform",
+				),
+				h.WithArgs("-log-level=debug"),
+			)
+
+			// check group.toml
+			tempGroupToml := filepath.Join(copyDir, "layers", "custom_group.toml")
+			groupContents, err := ioutil.ReadFile(tempGroupToml)
+			h.AssertNil(t, err)
+			h.AssertEq(t, len(groupContents) > 0, true)
+			var buildpackGroup lifecycle.BuildpackGroup
+			_, err = toml.Decode(string(groupContents), &buildpackGroup)
+			h.AssertNil(t, err)
+			h.AssertEq(t, buildpackGroup.Group[0].ID, "always_detect_buildpack")
+			h.AssertEq(t, buildpackGroup.Group[0].Version, "always_detect_buildpack_version")
+
+			// check plan.toml - should be empty
+			tempPlanToml := filepath.Join(copyDir, "layers", "custom_plan.toml")
```

Comment:
I kind of confused why this is empty...

Reply:
Because we're using `always_detect_order.toml` so there is no "actual plan".

Reply:
Oh, I see. Maybe it's worth expanding the comment, so that someone new to the code would not have to dig too much to understand it.

Reply:
I'll add some more explanation to the comment in my next commit. Please let me know if it is still not clear enough.

Reply:
Resolving this conversation for now but please feel free to reopen it.